### ISSUE TITLE
ipxe network somewhat broken when there is no 3GB ram limit

### ISSUE THREAD

**Issue by rgl** [pftf/RPi4#65]
When I remove the 3GB ram limit the ipxe network cannot be initialized.

To workaround this, it seems that using the ipxe `ifstat` command triggers a crash, but something (uefi?) is immediately restarting ipxe, and somehow it works the second time. Here's the output:

```plain
ESC (setup), F1 (shell), ENTER (boot).iPXE initialising devices...ok



iPXE 1.20.1 (g8f151) -- Open Source Network Boot Firmware -- http://ipxe.org
Features: DNS HTTP HTTPS iSCSI TFTP SRP EFI Menu
Platform..... efi
Architecture. arm64
Product...... Raspberry Pi 4 Model B
Manufacturer. Sony UK
Asset........ PI00000001
UUID......... 00d03114-0000-0000-0000-dca632b0ba1d
Serial....... 0000DCA632B0BA1D
Network interfaces:

  Error: Could not detect network connection.

  Error: Could not detect network connection.
iPXE initialising devices...ok



iPXE 1.20.1 (g8f151) -- Open Source Network Boot Firmware -- http://ipxe.org
Features: DNS HTTP HTTPS iSCSI TFTP SRP EFI Menu
Platform..... efi
Architecture. arm64
Product...... Raspberry Pi 4 Model B
Manufacturer. Sony UK
Asset........ PI00000001
UUID......... 00d03114-0000-0000-0000-dca632b0ba1d
Serial....... 0000DCA632B0BA1D
Network interfaces:
net0: dc:a6:32:b0:ba:1d using SNP on SNP-0x35c0c598 (closed)
  [Link:down, TX:0 TXE:0 RX:0 RXE:0]
  [Link status: Unknown (http://ipxe.org/1a086194)]
Configuring net0 dc:a6:32:b0:ba:1d (SNP) from dhcp...
Waiting for link-up on net0...... ok
Configuring [dhcp] (net0 dc:a6:32:b0:ba:1d)...... ok
```

**Comment by rgl** [comment]
Hummm everything works as expected when the memory is limited to 3GB.

**Comment by samerhaj** [comment]
I see link status/detection errors. 

We have some fixes for the GENET driver media detection code, that may be related.

**Comment by samerhaj** [comment]
Can you please pull this patch and see whether it resolves this issue or not?

https://edk2.groups.io/g/devel/message/61435

**Comment by samerhaj** [comment]
Fix committed to tianocore/edk2@2d233af64b8f73d1b1e138b302e6344f7c2e0f4e

Will port to RPi PlatformBootManagerLib and send a patch

**Comment by rgl** [comment]
I've tried v1.16 and the iPXE network initialization now works at the first. Thanks for looking into this! And, sorry for not being able to test this sooner.
